gulp/full.js

```diff
@@ -1,3 +1,15 @@
 var gulp = require('gulp');
+var gulpSequence = require('gulp-sequence');
 
-gulp.task('full', ['build', 'tslint', 'all-tests']);
+var isMaster = require('./util/is-master');
+var isTagAdded = require('./util/is-tag-added');
+
+gulp.task('build-and-tests', ['build', 'tslint', 'all-tests']);
+gulp.task('build-and-tests-and-publish', gulpSequence('build-and-tests', 'publish'));
+
+if (isMaster() && isTagAdded()) {
```

Comment:
I am thinking about moving this logic to https://github.com/gerich-home/it-depends/blob/master/appveyor.yml#L39
What do you think about it?